src/iconv.jl

```diff
@@ -102,11 +108,11 @@ function iconv_reset!(s::Union{StringEncoder, StringDecoder})
                 (Ptr{Void}, Ptr{Ptr{UInt8}}, Ref{Csize_t}, Ptr{Ptr{UInt8}}, Ref{Csize_t}),
                 s.cd, C_NULL, C_NULL, s.outbufptr, s.outbytesleft)
 
-    if ret == reinterpret(Csize_t, -1)
+    if ret == typemax(Csize_t)
```

Comment:
OK, so that would be a problem if `Int` and `Csize_t` are not of the same size. Better use `%` then?
```julia
julia> Int32(-1) % UInt64
0xffffffffffffffff

julia> Int32(-1) % UInt32
0xffffffff
```